e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,209 @@
+
+	_, err := val.ExecTx(ctx, nw.fiatTfRoles.Owner.KeyName(), "fiat-tokenfactory", "update-pauser", newPauser1.FormattedAddress())
+	require.NoError(t, err, "failed to broadcast update-pauser message")
+
+	showPauserRes, err := showPauser(ctx, val)
+	require.NoError(t, err, "failed to query show-pauser")
+	expectedGetPauserResponse := fiattokenfactorytypes.QueryGetPauserResponse{
+		Pauser: fiattokenfactorytypes.Pauser{
+			Address: string(newPauser1.FormattedAddress()),
+		},
+	}
+	require.Equal(t, expectedGetPauserResponse.Pauser, showPauserRes.Pauser)
+
+	unpauseFiatTF(t, ctx, val, newPauser1)
+
+	// ACTION: Update Pauser from non owner account
+	// EXPECTED: Request fails; pauser not updated
+	// Status:
+	// 	Pauser: newPauser1
+
+	w = interchaintest.GetAndFundTestUsers(t, ctx, "default", math.OneInt(), noble, noble)
+	newPauser2 := w[0]
+	alice := w[1]
+
+	_, err = val.ExecTx(ctx, alice.KeyName(), "fiat-tokenfactory", "update-pauser", newPauser2.FormattedAddress())
+	require.ErrorContains(t, err, "you are not the owner: unauthorized")
+
+	showPauserRes, err = showPauser(ctx, val)
+	require.NoError(t, err, "failed to query show-pauser")
+	require.Equal(t, expectedGetPauserResponse.Pauser, showPauserRes.Pauser)
+
+	// ACTION: Update Pauser from blacklisted owner account
+	// EXPECTED: Success; pauser updated
+	// Status:
+	// 	Pauser: newPauser1
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Owner)
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.Owner.KeyName(), "fiat-tokenfactory", "update-pauser", newPauser2.FormattedAddress())
+	require.NoError(t, err, "failed to broadcast update-pauser message")
+
+	showPauserRes, err = showPauser(ctx, val)
+	require.NoError(t, err, "failed to query show-pauser")
+	expectedGetPauserResponse = fiattokenfactorytypes.QueryGetPauserResponse{
+		Pauser: fiattokenfactorytypes.Pauser{
+			Address: string(newPauser2.FormattedAddress()),
+		},
+	}
+	require.Equal(t, expectedGetPauserResponse.Pauser, showPauserRes.Pauser)
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.Owner)
+
+	// ACTION: Update Pauser to blacklisted Pauser account
+	// EXPECTED: Success; pauser updated
+	// Status:
+	// 	Pauser: newPauser2
+
+	w = interchaintest.GetAndFundTestUsers(t, ctx, "new-pauser-3", math.OneInt(), noble)
+	newPauser3 := w[0]
+
+	blacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, newPauser3)
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.Owner.KeyName(), "fiat-tokenfactory", "update-pauser", newPauser3.FormattedAddress())
+	require.NoError(t, err, "failed to broadcast update-pauser message")
+
+	showPauserRes, err = showPauser(ctx, val)
+	require.NoError(t, err, "failed to query show-pauser")
+	expectedGetPauserResponse = fiattokenfactorytypes.QueryGetPauserResponse{
+		Pauser: fiattokenfactorytypes.Pauser{
+			Address: string(newPauser3.FormattedAddress()),
+		},
+	}
+	require.Equal(t, expectedGetPauserResponse.Pauser, showPauserRes.Pauser)
+}
```

Comment:
Could we add the happy path case as well? (i.e. a normal request — nothing is paused or blacklisted)

Reply:
Added in https://github.com/noble-assets/noble/pull/368/commits/a278e1e59be0319c50ef61a3a1ff79105c8fa702

Reply:
Added in: [`9a77cc4` (#369)](https://github.com/noble-assets/noble/pull/369/commits/9a77cc480b7a8b9d0534414f76a3f8325109702d)